### PR TITLE
🌟 Added link preview in the frontend

### PR DESCRIPTION
**Pull request by rezk2ll**
## Description

- adds the link preview 

## Related Issue

https://github.com/linagora/Twake/issues/2217

## Motivation and Context

- display previews for links detected in messages.

## How Has This Been Tested?

- locally

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/6764881/171865977-4d9955cd-8f2a-423b-8c37-5b4400c3c79a.mp4

image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added the Signed-off-by statement at the end of my commit message.

### CODE REVIEW

**Comment by RomaricMourgues** [comment]
Nice, I'll have some design change proposals later, I already have this ones:
- Reduce size of favicon (they seems blurry, I think default size is 16px)
- Align title first line with the favicon itself
- Add the domain name (like youtube.com) just under the title
- Reduce the title size (I think the same size as the description is ok, bold should be enough to differentiate title and description)
- If not already done, of course make it clickable, on slack the title is in blue and is clikable if we want to follow they implmentaiton

**Comment by rezk2ll** [comment]
updated:

https://user-images.githubusercontent.com/6764881/172189041-9afda088-0d7c-442e-937a-558357bdf1c4.mp4

image
image

**Comment by RomaricMourgues** [comment]
More change request:
- Let's put the domain next to the favicon instead of the title.
- Let's put the title aligned with the description
- Add border radius to the bottom on the link too

172189022-32a1cfbf-f072-4fd9-b9e1-1de818479b9c

File: twake/frontend/src/app/views/applications/messages/message/parts/MessageContent.tsx [hunk 1]
```diff
@@ -127,8 +128,13 @@ export default (props: Props) => {
...
 
-              {message?.files && (message?.files?.length || 0) > 0 && <MessageAttachments />}
-
+              {message?.files && (message?.files?.length || 0) > 0 ? (
+                <MessageAttachments />
+              ) : (
+                message?.links &&
+                (message?.links?.length || 0) > 0 &&
+                message.links.map(preview => <LinkPreview preview={preview} />)
+              )}
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/views/applications/messages/message/parts/MessageContent.tsx:137]: No that's wrong, a message can have both links and files actually

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]

**Pull request merged**